src/main/java/org/testng/internal/Invoker.java

```diff
@@ -1609,6 +1609,21 @@ public boolean isTrue(ITestNGMethod m, IClass c) {
     return result;
   }
 
+  private boolean isConfiguration(ITestNGMethod method) {
+    return method.isAfterClassConfiguration() || method.isBeforeClassConfiguration() ||
+        method.isBeforeGroupsConfiguration() || method.isAfterGroupsConfiguration() ||
+        method.isBeforeMethodConfiguration() || method.isAfterMethodConfiguration() ||
+        method.isBeforeTestConfiguration() || method.isAfterTestConfiguration() ||
+        method.isBeforeSuiteConfiguration() || method.isAfterSuiteConfiguration();
+  }
+
+  private boolean alreadyAdded(ITestNGMethod method, List<ITestNGMethod> methods) {
```

Comment:
Same: static + location.

About the naming, could be `containsConfigurationMethod(List<> methods, ITest.. method)`